src/overlays/actors/ovl_Obj_Purify/z_obj_purify.c

```diff
@@ -27,42 +56,256 @@ const ActorInit Obj_Purify_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+ObjPurifyInfo ObjPurifyInfoList[] = {
+    {
+        OBJECT_NUMA_OBJ,
+        0.1f,
+        { NULL, NULL },
+        { object_numa_obj_DL_0128E0, object_numa_obj_DL_012D90 },
+        { object_numa_obj_Matanimheader_012CF8, object_numa_obj_Matanimheader_0131A8 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+    {
+        OBJECT_DEKUCITY_OBJ,
+        1.0f,
+        { NULL, NULL },
+        { object_dekucity_obj_DL_0004E0, object_dekucity_obj_DL_001030 },
+        { object_dekucity_obj_Matanimheader_000E28, object_dekucity_obj_Matanimheader_001978 },
+        &object_dekucity_obj_Colheader_0002AC,
+        1,
+    },
+    {
+        OBJECT_DEKUCITY_OBJ,
+        1.0f,
+        { object_dekucity_obj_DL_001ED8, object_dekucity_obj_DL_003998 },
+        { object_dekucity_obj_DL_001D80, NULL },
+        { object_dekucity_obj_Matanimheader_003820, NULL },
+        NULL,
+        1,
+    },
+    {
+        OBJECT_NUMA_OBJ,
+        0.1f,
+        { NULL, NULL },
+        { object_numa_obj_DL_012A60, object_numa_obj_DL_012F10 },
+        { object_numa_obj_Matanimheader_012B58, object_numa_obj_Matanimheader_013008 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+    {
+        OBJECT_NUMA_OBJ,
+        0.1f,
+        { NULL, NULL },
+        { object_numa_obj_DL_012BF0, object_numa_obj_DL_0130A0 },
+        { object_numa_obj_Matanimheader_012CE8, object_numa_obj_Matanimheader_013198 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+};
+
+void ObjPurify_SetSysMatrix(f32 ypos) {
+    Matrix_GetCurrent()->yw = ypos;
+}
+
+s32 ObjPurify_IsPurified(ObjPurify* this) {
+    ObjPurifyInfo* info;
+    info = &ObjPurifyInfoList[OBJPURIFY_GET_INFO_INDEX(this)];
```

Comment:
```suggestion
    ObjPurifyInfo* info = &ObjPurifyInfoList[OBJPURIFY_GET_INFO_INDEX(this)];
```
You can combine these lines.